src/recipe.rs

```diff
@@ -16,6 +16,76 @@ fn error_from_signal(recipe: &str, line_number: Option<usize>, exit_status: Exit
   }
 }
 
+#[derive(Debug)]
+struct SystemMap {
+  windows: bool,
+  macos: bool,
+  linux: bool,
+  openbsd: bool,
+  unix: bool,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum System {
+  Windows,
+  MacOS,
+  Linux,
+  OpenBSD,
+  Unix,
+}
+
+impl System {
+  fn current() -> System {
+    use System::*;
+    if cfg!(target_os = "linux") {
+      return Linux;
+    }
+    if cfg!(target_os = "openbsd") {
+      return OpenBSD;
+    }
+    if cfg!(target_os = "macos") {
+      return MacOS;
+    }
+    if cfg!(target_os = "windows") || cfg!(windows) {
+      return Windows;
+    }
+    if cfg!(unix) {
+      return Unix;
+    }
+    panic!("No recognized system");
+  }
+
+  fn enabled(self, enabled: SystemMap, disabled: SystemMap) -> bool {
+    match self {
+      System::Windows => {
+        !disabled.windows
+          && (enabled.windows
+            || !(enabled.macos || enabled.linux || enabled.openbsd || enabled.unix))
+      }
+      System::MacOS => {
```

Comment:
On MacOS, and other unix systems, a recipe with `[not(unix)]` will be enabled. This seems wrong to me.